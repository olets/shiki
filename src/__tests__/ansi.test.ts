import { getHighlighter } from '../index'

test('renders ansi to html with default theme', async () => {
  const highlighter = await getHighlighter({ theme: 'monokai' })

  const out = highlighter.ansiToHtml(`[0;30;43mâ€‰WARNâ€‰[0m using --force I sure hope you know what you are doing
Scope: all 6 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: [0;32m+952[0m
[0;32m++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++[0m
Progress: resolved [0;104m952[0m, reused [0;104m910[0m, downloaded [0;104m42[0m, added [0;104m952[0m, done
Done in 15.7s`)

  expect(out).toMatchSnapshot()
})

test('renders ansi to html with provided theme', async () => {
  const highlighter = await getHighlighter({ themes: ['monokai', 'nord'] })

  const out = highlighter.ansiToHtml(
    `[0;30;43mâ€‰WARNâ€‰[0m using --force I sure hope you know what you are doing
Scope: all 6 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: [0;32m+952[0m
[0;32m++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++[0m
Progress: resolved [0;104m952[0m, reused [0;104m910[0m, downloaded [0;104m42[0m, added [0;104m952[0m, done
Done in 15.7s`,
    { theme: 'nord' }
  )

  expect(out).toMatchSnapshot()
})

test('renders ansi to tokens', async () => {
  const highlighter = await getHighlighter({ theme: 'monokai' })

  const out =
    highlighter.ansiToThemedTokens(`[0;30;43mâ€‰WARNâ€‰[0m using --force I sure hope you know what you are doing
Scope: all 6 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: [0;32m+952[0m
[0;32m++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++[0m
Progress: resolved [0;104m952[0m, reused [0;104m910[0m, downloaded [0;104m42[0m, added [0;104m952[0m, done
Done in 15.7s`)

  expect(out).toMatchSnapshot()
})

test('renders ansi to html with css-variables theme', async () => {
  const highlighter = await getHighlighter({
    theme: 'css-variables'
  })

  const allAnsiStyles = `
[30mblack[0m
[30;2mblack (dim)[0m
[31mred[0m
[31;2mred (dim)[0m
[32mgreen[0m
[32;2mgreen (dim)[0m
[33myellow[0m
[33;2myellow (dim)[0m
[34mblue[0m
[34;2mblue (dim)[0m
[35mmagenta[0m
[35;2mmagenta (dim)[0m
[36mcyan[0m
[36;2mcyan (dim)[0m
[37mwhite[0m
[37;2mwhite (dim)[0m
[90mbrightBlack[0m
[90;2mbrightBlack (dim)[0m
[91mbrightRed[0m
[91;2mbrightRed (dim)[0m
[92mbrightGreen[0m
[92;2mbrightGreen (dim)[0m
[93mbrightYellow[0m
[93;2mbrightYellow (dim)[0m
[94mbrightBlue[0m
[94;2mbrightBlue (dim)[0m
[95mbrightMagenta[0m
[95;2mbrightMagenta (dim)[0m
[96mbrightCyan[0m
[96;2mbrightCyan (dim)[0m
[97mbrightWhite[0m
[97;2mbrightWhite (dim)[0m
[1mbold[0m
[3mitalic[0m
[3munderline[0m
[9mstrikethrough[0m
`

  const out = highlighter.ansiToHtml(allAnsiStyles)

  expect(out).toMatchSnapshot()
})
