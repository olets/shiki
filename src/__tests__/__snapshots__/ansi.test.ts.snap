// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ansi to html with css-variables theme 1`] = `
"<pre class=\\"shiki css-variables\\" style=\\"background-color: var(--shiki-color-background)\\" tabindex=\\"0\\"><code><span class=\\"line\\"></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-black)\\">black</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-black-dim)\\">black (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-red)\\">red</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-red-dim)\\">red (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-green)\\">green</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-green-dim)\\">green (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-yellow)\\">yellow</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-yellow-dim)\\">yellow (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-blue)\\">blue</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-blue-dim)\\">blue (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-magenta)\\">magenta</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-magenta-dim)\\">magenta (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-cyan)\\">cyan</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-cyan-dim)\\">cyan (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-white)\\">white</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-white-dim)\\">white (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-black)\\">brightBlack</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-black-dim)\\">brightBlack (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-red)\\">brightRed</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-red-dim)\\">brightRed (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-green)\\">brightGreen</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-green-dim)\\">brightGreen (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-yellow)\\">brightYellow</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-yellow-dim)\\">brightYellow (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-blue)\\">brightBlue</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-blue-dim)\\">brightBlue (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-magenta)\\">brightMagenta</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-magenta-dim)\\">brightMagenta (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-cyan)\\">brightCyan</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-cyan-dim)\\">brightCyan (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-white)\\">brightWhite</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-ansi-bright-white-dim)\\">brightWhite (dim)</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-text); font-weight: bold\\">bold</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-text); font-style: italic\\">italic</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-text); font-style: italic\\">underline</span></span>
<span class=\\"line\\"><span style=\\"color: var(--shiki-color-text)\\">strikethrough</span></span>
<span class=\\"line\\"></span></code></pre>"
`;

exports[`renders ansi to html with default theme 1`] = `
"<pre class=\\"shiki monokai\\" style=\\"background-color: #272822\\" tabindex=\\"0\\"><code><span class=\\"line\\"><span style=\\"color: #333333\\"> WARN </span><span style=\\"color: #F8F8F2\\"> using --force I sure hope you know what you are doing</span></span>
<span class=\\"line\\"><span style=\\"color: #F8F8F2\\">Scope: all 6 workspace projects</span></span>
<span class=\\"line\\"><span style=\\"color: #F8F8F2\\">Lockfile is up to date, resolution step is skipped</span></span>
<span class=\\"line\\"><span style=\\"color: #F8F8F2\\">Packages: </span><span style=\\"color: #86B42B\\">+952</span></span>
<span class=\\"line\\"><span style=\\"color: #86B42B\\">++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++</span></span>
<span class=\\"line\\"><span style=\\"color: #F8F8F2\\">Progress: resolved </span><span style=\\"color: #F8F8F2\\">952</span><span style=\\"color: #F8F8F2\\">, reused </span><span style=\\"color: #F8F8F2\\">910</span><span style=\\"color: #F8F8F2\\">, downloaded </span><span style=\\"color: #F8F8F2\\">42</span><span style=\\"color: #F8F8F2\\">, added </span><span style=\\"color: #F8F8F2\\">952</span><span style=\\"color: #F8F8F2\\">, done</span></span>
<span class=\\"line\\"><span style=\\"color: #F8F8F2\\">Done in 15.7s</span></span></code></pre>"
`;

exports[`renders ansi to html with provided theme 1`] = `
"<pre class=\\"shiki nord\\" style=\\"background-color: #2e3440ff\\" tabindex=\\"0\\"><code><span class=\\"line\\"><span style=\\"color: #3b4252\\"> WARN </span><span style=\\"color: #d8dee9ff\\"> using --force I sure hope you know what you are doing</span></span>
<span class=\\"line\\"><span style=\\"color: #d8dee9ff\\">Scope: all 6 workspace projects</span></span>
<span class=\\"line\\"><span style=\\"color: #d8dee9ff\\">Lockfile is up to date, resolution step is skipped</span></span>
<span class=\\"line\\"><span style=\\"color: #d8dee9ff\\">Packages: </span><span style=\\"color: #a3be8c\\">+952</span></span>
<span class=\\"line\\"><span style=\\"color: #a3be8c\\">++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++</span></span>
<span class=\\"line\\"><span style=\\"color: #d8dee9ff\\">Progress: resolved </span><span style=\\"color: #d8dee9ff\\">952</span><span style=\\"color: #d8dee9ff\\">, reused </span><span style=\\"color: #d8dee9ff\\">910</span><span style=\\"color: #d8dee9ff\\">, downloaded </span><span style=\\"color: #d8dee9ff\\">42</span><span style=\\"color: #d8dee9ff\\">, added </span><span style=\\"color: #d8dee9ff\\">952</span><span style=\\"color: #d8dee9ff\\">, done</span></span>
<span class=\\"line\\"><span style=\\"color: #d8dee9ff\\">Done in 15.7s</span></span></code></pre>"
`;

exports[`renders ansi to tokens 1`] = `
Array [
  Array [
    Object {
      "color": "#333333",
      "content": " WARN ",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": " using --force I sure hope you know what you are doing",
      "fontStyle": 0,
    },
  ],
  Array [
    Object {
      "color": "#F8F8F2",
      "content": "Scope: all 6 workspace projects",
      "fontStyle": 0,
    },
  ],
  Array [
    Object {
      "color": "#F8F8F2",
      "content": "Lockfile is up to date, resolution step is skipped",
      "fontStyle": 0,
    },
  ],
  Array [
    Object {
      "color": "#F8F8F2",
      "content": "Packages: ",
      "fontStyle": 0,
    },
    Object {
      "color": "#86B42B",
      "content": "+952",
      "fontStyle": 0,
    },
  ],
  Array [
    Object {
      "color": "#86B42B",
      "content": "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++",
      "fontStyle": 0,
    },
  ],
  Array [
    Object {
      "color": "#F8F8F2",
      "content": "Progress: resolved ",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": "952",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": ", reused ",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": "910",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": ", downloaded ",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": "42",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": ", added ",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": "952",
      "fontStyle": 0,
    },
    Object {
      "color": "#F8F8F2",
      "content": ", done",
      "fontStyle": 0,
    },
  ],
  Array [
    Object {
      "color": "#F8F8F2",
      "content": "Done in 15.7s",
      "fontStyle": 0,
    },
  ],
]
`;
